qa-tests-backend/src/test/groovy/ImageSignaturesTest.groovy

```diff
@@ -0,0 +1,270 @@
+            .setName("with-signature-verified-by-tekton")
+            .setImage("gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/git-init@" +
+                    "sha256:79f768d28ff9af9fcbf186f9fc1b8e9f88835dfb07be91610a1f17cf862db89e")
+            .addLabel("app", "image-with-signature-tekton-test")
+            .setCommand(["/bin/sh", "-c", "/bin/sleep 600"])
+            .setNamespace(SIGNATURE_TESTING_NAMESPACE)
+
+    // Deployment holding an image which has a cosign signature that is not verifiable by any cosign public key.
+    static final private Deployment UNVERIFIABLE_DEPLOYMENT = new Deployment()
+            .setName("with-signature-unverifiable")
+            .setImage("docker.io/istio/proxyv2@sha256:134e99aa9597fdc17305592d13add95e2032609d23b4c508bd5ebd32ed2df47d")
+            .addLabel("app", "image-with-unverifiable-signature-test")
+            .setCommand(["/usr/local/bin/pilot-agent", "wait", "--timeoutSeconds", "6000"])
+            .setNamespace(SIGNATURE_TESTING_NAMESPACE)
+
+    // Deployment holding an image which does not have a cosign signature.
+    static final private Deployment WITHOUT_SIGNATURE_DEPLOYMENT = new Deployment()
+            .setName("without-signature")
+            .setImage("docker.io/nginx@sha256:63aa22a3a677b20b74f4c977a418576934026d8562c04f6a635f0e71e0686b6d")
+            .addLabel("app", "image-without-signature")
+            .setNamespace(SIGNATURE_TESTING_NAMESPACE)
+
+    // List of deployments used within the tests. This will be used during setup of the spec / teardown to create /
+    // delete all deployments.
+    // NOTE: If you add another deployment, make sure to add it here as well.
+    static final private List<Deployment> DEPLOYMENTS = [
+            DISTROLESS_DEPLOYMENT,
+            TEKTON_DEPLOYMENT,
+            UNVERIFIABLE_DEPLOYMENT,
+            WITHOUT_SIGNATURE_DEPLOYMENT,
+    ]
+
+    // Base policy which will be used for creating subsequent policies that have signature integration IDs as values.
+    static final private Policy.Builder BASE_POLICY = Policy.newBuilder()
+            .addLifecycleStages(PolicyOuterClass.LifecycleStage.DEPLOY)
+            .addCategories("Test")
+            .setDisabled(false)
+            .setSeverityValue(2)
+            .addAllScope([SIGNATURE_TESTING_NAMESPACE].collect
+                    { ScopeOuterClass.Scope.newBuilder().setNamespace(it).build() })
+
+    @Shared
+    final private List<String> createdPolicyIds = []
+
+    @Shared
+    final private Map<String, String> createdSignatureIntegrations = [:]
+
+    def setupSpec() {
+        orchestrator.createNamespace(SIGNATURE_TESTING_NAMESPACE)
+
+        // Signature integration "Distroless" which holds only the distroless cosign public key.
+        String distrolessSignatureIntegrationID = createSignatureIntegration(
+                DISTROLESS, DISTROLESS_PUBLIC_KEY
+        )
+        assert distrolessSignatureIntegrationID
```

Comment:
What if this assertion fails? Will `cleanupSpec` be called after that?

Reply:
Yes, `cleanupSepc` will be called there (I had typos in the beginning within the deployment, so I can safely say that :) )